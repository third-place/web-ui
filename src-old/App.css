body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
  'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
  sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 18pt;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
  monospace;
}

.App {
  position: relative;
}

.App-logo {
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  height: 70px;
  display: flex;
  justify-content: space-between;
  background-color: #011627;
  color: #ffffff;
}

div.navigation {
  align-self: center;
  padding: 20px;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

div.body {
  padding: 0 20px;
}

div.row {
    margin: 10px;
}

div.post {
  overflow-x: scroll;
  overflow-y: hidden;
}

div.post img {
  max-width: 100%;
}

a {
  text-decoration: none !important;
  color: #e71d36;
}

h2 {
  margin: 10px 0 20px;
}

h5 {
  margin: 0 !important;
}

p {
  margin: 5px 0;
}

pre {
  background: #f4f4f4;
  border: 1px solid #ddd;
  border-left: 3px solid #e71d36;
  color: #666;
  page-break-inside: avoid;
  font-family: monospace;
  font-size: 15px;
  line-height: 1.6;
  margin-bottom: 1.6em;
  max-width: 100%;
  overflow: auto;
  padding: 1em 1.5em;
  display: block;
  word-wrap: break-word;
}

.MuiTypography-h6 {
  font-weight: normal !important;
}

.post-gallery {
  max-width: 200px;
  max-height: 100px;
  border-radius: 10px;
  padding: 4px;
}

.MuiTypography-body2 {
  padding: 10px 0;
  font-size: 14pt !important;
}
