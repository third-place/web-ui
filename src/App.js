import { ThemeProvider } from '@emotion/react';
import { useEffect, useState } from 'react';
import { getFollowers, getFollowing } from './actions/follow';
import { getNotifications } from './actions/notification';
import { getPostsForUser } from './actions/post';
import { getUser, refreshSession } from './actions/session';
import ProtectedRoute from './components/ProtectedRoute';
import Album from './pages/Album';
import Drafts from './pages/Drafts';
import Followers from './pages/Followers';
import Following from './pages/Following';
import Image from './pages/Image';
import InviteList from './pages/InviteList';
import ModerateUser from './pages/ModerateUser';
import Notifications from './pages/Notifications';
import OTP from './pages/OTP';
import PostEdit from './pages/PostEdit';
import UpdateProfile from './pages/UpdateProfile';
import {
  User,
  Home,
  Post,
} from './pages';
import './App.css';
import Login from './pages/Login';
import ForgotPassword from './pages/ForgotPassword';
import Signup from './pages/Signup';
import Context from './utils/Context';
import {
  BrowserRouter as Router,
  Routes,
  Route,
} from "react-router-dom";
import { darkTheme, lightTheme } from './utils/theme';

function App() {
  const [sessionToken, setSessionToken] = useState(localStorage.getItem("token"));
  const [loggedInUser, setLoggedInUser] = useState(null);
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [userEmail, setUserEmail] = useState(null);
  const [isAppLoaded, setIsAppLoaded] = useState(false);
  const [posts, setPosts] = useState([]);
  const [follows, setFollows] = useState([]);
  const [followers, setFollowers] = useState([]);
  const [notifications, setNotifications] = useState([]);
  const [uiMode, setUiMode] = useState(localStorage.getItem("uiMode") ?? "dark");

  const tryGetNotifications = async (token) => {
    const response = await getNotifications(token);
    const data = await response.json();
    setNotifications(data);
  };

  const appContext = {
    sessionToken,
    setSessionToken,
    loggedInUser,
    setLoggedInUser,
    userEmail,
    setUserEmail,
    isAppLoaded,
    posts,
    setPosts,
    follows,
    setFollows,
    followers,
    setFollowers,
    isLoggedIn,
    setIsLoggedIn,
    uiMode,
    setUiMode,
    notifications,
    tryGetNotifications,
  };

  const tryGetUser = async (token) => {
    const response = await getUser(token);
    const data = await response.json();
    setLoggedInUser(data.user);
    setIsLoggedIn(true);
    setIsAppLoaded(true);
    return tryGetNotifications(token);
  };

  const tryRefreshSession = async (token) => {
    const response = await refreshSession(token);
    const data = await response.json();
    setSessionToken(data.token);
    localStorage.setItem("token", data.token);
    return data.token
  };

  const reloadFollowing = async () => {
    const response = await getFollowing(loggedInUser.username);
    const data = await response.json();
    setFollows(data);
  };

  const reloadFollowers = async () => {
    const response = await getFollowers(loggedInUser.username);
    const data = await response.json();
    setFollowers(data);
  };

  const setLoggedOutState = async () => {
    setLoggedInUser(null);
    setIsLoggedIn(false);
    setIsAppLoaded(true);
    const response = await getPostsForUser(null, 'dan');
    const data = await response.json();
    setPosts(data);
  };

  useEffect(() => {
    (async () => {
      if (!sessionToken) {
        console.log("no session found");
        await setLoggedOutState();
        return;
      }
      try {
        await tryGetUser(sessionToken);
      } catch (e) {
        try {
          const refreshToken = await tryRefreshSession(sessionToken);
          await tryGetUser(refreshToken);
        } catch (e) {
          await setLoggedOutState();
        }
      }
    })();
  }, []);

  useEffect(() => {
    if (loggedInUser) {
      (async function () {
        await reloadFollowing();
        await reloadFollowers();
      })();
    }
  }, [isLoggedIn]);

  return (
    <ThemeProvider theme={uiMode === "light" ? lightTheme : darkTheme}>
      <Context.Provider value={appContext}>
        <Router>
          <Routes>
            <Route path="/u/:username" element={<User />} />
            <Route path="/u/:username/following" element={<Following />} />
            <Route path="/u/:username/followers" element={<Followers />} />
            <Route path="/i/:uuid" element={<Image />} />
            <Route path="/p/:uuid" element={<Post />} />
            <Route path="/p-edit/:uuid" element={<PostEdit />} />
            <Route path="/a/:uuid" element={<Album />} />
            <Route path="/drafts" element={<Drafts />} />
            <Route path="/login" element={<Login />} />
            <Route path="/signup" element={<Signup />} />
            <Route path="/otp" element={<OTP />} />
            <Route path="/notifications" element={<Notifications />} />
            <Route path="/invite" element={<ProtectedRoute component={InviteList} role="admin" />} />
            <Route path="/forgot-password" element={<ForgotPassword />} />
            <Route path="/update-profile" element={<ProtectedRoute component={UpdateProfile} />} />
            <Route path="/moderate-user/:username" element={<ProtectedRoute role="moderator" component={ModerateUser} />} />
            <Route path="/moderate-post/:uuid" element={<ProtectedRoute role="moderator" component={ModerateUser} />} />
            <Route path="/" element={<Home />} />
          </Routes>
        </Router>
      </Context.Provider>
    </ThemeProvider>
  );
}

export default App;
