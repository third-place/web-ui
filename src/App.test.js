import renderer from 'react-test-renderer';
import App from './App';

test('renders correctly', () => {
  // when
  const tree = renderer
    .create(<App />)
    .toJSON();

  // then
  expect(tree).toMatchSnapshot();
});
