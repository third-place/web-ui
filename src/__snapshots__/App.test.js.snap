// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-k008qs"
>
  <div
    className="MuiDrawer-root MuiDrawer-docked css-1o2n5ug-MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-15q07m6-MuiPaper-root-MuiDrawer-paper"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      />
      <hr
        className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <div
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-p5wds0-MuiButtonBase-root-MuiListItem-root"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root css-19g8jym-MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jxdcj3-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-5kyoyc-MuiTypography-root"
            >
              Home
            </span>
          </div>
        </div>
        <div>
          <div
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-p5wds0-MuiButtonBase-root-MuiListItem-root"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root css-19g8jym-MuiListItemIcon-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jxdcj3-MuiSvgIcon-root"
                data-testid="LoginIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-5kyoyc-MuiTypography-root"
              >
                Login
              </span>
            </div>
          </div>
          <div
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-p5wds0-MuiButtonBase-root-MuiListItem-root"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root css-19g8jym-MuiListItemIcon-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jxdcj3-MuiSvgIcon-root"
                data-testid="FollowTheSignsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9.5 5.5c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zM5.75 8.9 3 23h2.1l1.75-8L9 17v6h2v-7.55L8.95 13.4l.6-3C10.85 12 12.8 13 15 13v-2c-1.85 0-3.45-1-4.35-2.45l-.95-1.6C9.35 6.35 8.7 6 8 6c-.25 0-.5.05-.75.15L2 8.3V13h2V9.65l1.75-.75M13 2v7h3.75v14h1.5V9H22V2h-9zm5.01 6V6.25H14.5v-1.5h3.51V3l2.49 2.5L18.01 8z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-5kyoyc-MuiTypography-root"
              >
                Signup
              </span>
            </div>
          </div>
        </div>
      </ul>
    </div>
  </div>
  <main
    className="MuiBox-root css-1kxwji3"
  />
</div>
`;
